framework/include/utils/RankTwoTensor.h

```diff
@@ -169,10 +187,23 @@ class RankTwoTensorTempl : public TensorValue<T>
    */
   void fillFromScalarVariable(const VariableValue & scalar_variable);
 
-public:
   /// returns _coords[i][c], ie, column c, with c = 0, 1, 2
   TypeVector<T> column(const unsigned int c) const;
 
+  /// multiply vector v with row n of this tensor
+  T rowMultiply(std::size_t n, const TypeVector<T> & v) const;
+
+  /// return the marix multiplied with its transpose A*A^T (guaranteed symmetric)
+  static RankTwoTensorTempl<T> timesTranspose(const RankTwoTensorTempl<T> &);
```

Comment:
I recommend adding also a `transposeTimes` method.

Reply:
In the classes I converted thus far this has not occurred yet.